crates/driver/openapi.yml

```diff
@@ -76,8 +154,107 @@ components:
           quote. For example, this Solver might not be aware of any liquidity between the tokens.
         type: object
         properties:
-          unfillable_reason:
+          unfillableReason:
             type: string
+    SolveRequest:
+      description: Request to the solve endpoint.
+      type: object
+      properties:
+        auction:
+          $ref: "#/components/schemas/Auction"
+        deadline:
+          description: |
+            The time until which the caller expects a response.
+
+            Encoded as ISO 8601 UTC.
+          type: string
+          example: "2020-12-03T18:35:18.814523Z"
+    SolveResponse:
+      description: Response of the solve endpoint.
+      type: object
+      properties:
+        objective:
+          description: |
+            The objective value of the solution.
+
+            `null` indicates that solving was successful but no solution was found.
+          type: number
+          nullable: true
+        signature:
+          description: |
+            Signature confirming that the Solver promised to have this solution for this auction.
+
+            TODO: Decide what is signed and how it is signed.
+          type: string
+    ExecuteRequest:
+      description: Request to the execute endpoint.
+      type: object
+      properties:
+        auctionId:
+          description: Id of the auction that should be executed.
+          type: integer
```

Comment:
To enshrine more of the commitment the solver is supposed to make maybe it makes sense to request to execute the signature they sent instead of the `auctionId`? But I guess it's implicit that we expect the final solution to be the same they promised during the `/solve` request.

Reply:
This is a nice idea. I think we could also send the signature back with the auction data (they would be able to verify locally and respond with a 403 for example if the signature is bad).

I would leave this out for the moment (as we are also leaving out signatures from the `/solve`) and decide when we add signatures to the solver API.

Reply:
Yeah, makes sense to only consider this change once we know what our signatures should look like in the first place.